hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/storage/row/TestHoodieInternalRowParquetWriter.java

```diff
@@ -123,6 +125,45 @@ public void testProperWriting(boolean parquetWriteLegacyFormatEnabled) throws Ex
     });
   }
 
+  @Test
+  @Timeout(value = 60)
+  public void testFileSizeCheck() throws Exception {
+    HoodieWriteConfig.Builder writeConfigBuilder =
+        SparkDatasetTestUtils.getConfigBuilder(basePath, timelineServicePort);
+
+    HoodieRowParquetWriteSupport writeSupport = getWriteSupport(writeConfigBuilder, hadoopConf, true);
+    HoodieWriteConfig cfg = writeConfigBuilder.build();
+
+    // Make the written records could exceed maxRowCountForSizeCheck
+    cfg.setValue(HoodieStorageConfig.PARQUET_MAX_FILE_SIZE, String.valueOf(5 * 1024 * 1024));
+    HoodieParquetConfig<HoodieRowParquetWriteSupport> parquetConfig = new HoodieParquetConfig<>(writeSupport,
+        CompressionCodecName.SNAPPY, cfg.getParquetBlockSize(), cfg.getParquetPageSize(), cfg.getParquetMaxFileSize(),
+        writeSupport.getHadoopConf(), cfg.getParquetCompressionRatio(), cfg.parquetDictionaryEnabled());
+
+    Path filePath = new Path(basePath + "/check_file_size.parquet");
+
+    try (HoodieInternalRowParquetWriter writer = new HoodieInternalRowParquetWriter(filePath, parquetConfig)) {
+
+      while (writer.canWrite()) {
+        Dataset<Row> inputRows = SparkDatasetTestUtils.getRandomRows(sqlContext, 100,
```

Comment:
Can we make the max file size small enough and try to remove the while loop, it is risky to write while loop in testing which would preempt CPU time slice

Reply:
If we make the max file size too small, we cannot ensure the written records reach `maxRowCountForSizeCheck`, whereas we cannot test the logic of `DEFAULT_MAXIMUM_RECORD_COUNT_FOR_CHECK`, so here we need to make the maxFileSize larger.

```
Math.min(
          // Do check it in the halfway
          Math.max(DEFAULT_MINIMUM_RECORD_COUNT_FOR_CHECK, (maxFileSize / avgRecordSize - writtenCount) / 2),
          DEFAULT_MAXIMUM_RECORD_COUNT_FOR_CHECK)
``

I test it several times, the test usually finished in 8 - 12 sec, maybe it doesn't hurt too much for the performance?

Reply:
@danny0405  Hi, I change the logic here to generate more records for one batch, now the while loop only iterate 2 times, and the time reduced to 5-6 secs.

Reply:
I would rather we write a unit test for this file size decision only, there is even no need to write real files.

Reply:
Sure, let me try to fix it.